src/game/Spells/Scripts/Scripting/ClassScripts/Paladin.cpp

```diff
@@ -239,6 +239,33 @@ struct JudgementsOfTheWiseEnergize : public SpellScript
     }
 };
 
+struct ArdentDefender : public AuraScript
+{
+    void OnAbsorb(Aura* aura, int32& currentAbsorb, int32& remainingDamage, uint32& /*reflectedSpellId*/, int32& /*reflectDamage*/, bool& preventedDeath) const override
+    {
+        currentAbsorb = 0;
+        Player* player = dynamic_cast<Player*>(aura->GetCaster());
+        if (!player)
+            return;
+        if (int32(player->GetHealth()) - remainingDamage > (player->GetMaxHealth() * 0.35))
+            return;
+        int32 reduction = 0;
+        int32 healMod = 0;
+        if (Aura* aur = aura->GetHolder()->GetAuraByEffectIndex(SpellEffectIndex(EFFECT_INDEX_0)))
+            reduction = aur->GetBasePoints();
+        if (Aura* aur = aura->GetHolder()->GetAuraByEffectIndex(SpellEffectIndex(EFFECT_INDEX_1)))
+            healMod = aur->GetBasePoints();
+        remainingDamage *= (100 - reduction) / 100.f;
+        if (int32(player->GetHealth()) - remainingDamage > 0 || player->HasAura(66233))
+            return;
+        healMod = player->GetMaxHealth() * (healMod / 100.f);
+        remainingDamage = 0;
+        // ToDo: "(amount healed based on defense)" find out ratios etc.
```

Comment:
![image](https://user-images.githubusercontent.com/3718129/206850124-f379dd49-a3b3-4371-880d-0fe91a2b88c4.png)
this is the only thing I could find and it's not an official resource